### ISSUE TITLE
Fix `selector-pseudo-element-no-unknown` false positives for `view-transition` pseudo-elements

### ISSUE THREAD

**Issue by danielroe** [stylelint/stylelint#7069]
### What minimal example or steps are needed to reproduce the bug?

```css
::view-transition-old(menu) {
  transform: rotate(0deg);
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-pseudo-element-no-unknown": true
  }
}
```

### How did you run Stylelint?

https://stylelint.io/demo/#N4Igxg9gJgpiBcD4DcCWMDuBaALgJwEMA7AZ1R1QiKwgBsoAKAWxiIFcBKAAmAB0iuXfMRIAzCHibwueCDgI4YDAAywA5hwDc-AL78QAGhCjUtGADkCLBCBgAPKwAczAOjAkSh8FRNqbfAS5eWztFIigSYOkAbX5BQWCSHABPM1pUIhwsSCJfLCTiKAI8KGC4rgBdXS8c3wAxCSYFGwArEiovWEdPRACEkCTUmHTMqKCQWgUYJOCDcsSUtIys2tQ1fPlw4tKEccnFGZBqnSA

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "15.10.1"
}
```

### What did you expect to happen?

No problems to be reported.

### What actually happened?

The following problems were reported:

```
 172:1  ✖  Unexpected unknown pseudo-element selector "::view-transition-old"  selector-pseudo-element-no-unknown
```

### Do you have a proposal to fix the bug?

Add the new pseudo element selectors to the list, alongside other experimental ones like `target-text` and `spelling-error`. **Note**: I've reported this as a bug but it may of course be a feature, depending on how you think about it.

Documentation: https://developer.mozilla.org/en-US/docs/Web/CSS/::view-transition

**Comment by jeddy3** [comment]
@danielroe Thank you for the clear report, using the template and for the pull request.

> Note: I've reported this as a bug but it may of course be a feature, depending on how you think about it.

You were right to as we typical class these as false positive bugs.
